### PR TITLE
[Enhancement] Add error handler when getting host failed

### PR DESCRIPTION
**Pull request by del-zhenwu**
Signed-off-by: del-zhenwu <zhenxiang.li@zilliz.com>

## Motivation

Try to fix issue #1339, add exception handler when get host failed

## Modification

Add exception handler in mmcv/runner/utils.py

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/runner/utils.py [hunk 1]
```diff
@@ -13,7 +14,13 @@
 
 
 def get_host_info():
-    return f'{getuser()}@{gethostname()}'
+    host = ''
```
Comment (zhouzaida) [inline mmcv/runner/utils.py:17]: Suggest adding a comment to explain what situation the host can not be obtained

File: mmcv/runner/utils.py [hunk 2]
```diff
@@ -13,7 +14,18 @@
 
 
 def get_host_info():
-    return f'{getuser()}@{gethostname()}'
+    """Get hostname and username.
+
+    Return empty string if exception raised, e.g. `getuser()` will lead to
+    error in docker container
```
Comment (zhouzaida) [inline mmcv/runner/utils.py:20]: Maybe we can also mount `/etc/passwd` from our host when we run the docker with the following command to fix the exception. More details at https://stackoverflow.com/questions/48527958/docker-error-getting-username-from-password-database

```bash
docker run -v /etc/passwd:/etc/passwd ...
```

Comment (del-zhenwu) [reply mmcv/runner/utils.py:20]: > Maybe we can also mount `/etc/passwd` from our host when we run the docker with the following command to fix the exception. More details at https://stackoverflow.com/questions/48527958/docker-error-getting-username-from-password-database
> 
> ```shell
> docker run -v /etc/passwd:/etc/passwd ...
> ```

This should be work, but from the user's point of view, it is even more inconvenient

**Review by zhouzaida — APPROVED** [review]

File: mmcv/runner/utils.py [hunk 3]
```diff
@@ -13,7 +14,18 @@
...
+    """
+    host = ''
+    try:
+        host = f'{getuser()}@{gethostname()}'
+    except Exception as e:
+        warnings.warn('Host or user not found: %s' % (str(e)))
```
Comment (ZwwWayne) [inline mmcv/runner/utils.py:26]: We usually use f-string, so please change it to `f'Host or user not found: {str(e)}'`

**Pull request merged**